### DISCUSSION TITLE
save the console output from DotNetCoreExecute

### DISCUSSION THREAD

**Discussion by BartNetJS** [cake-build/cake#3202] [Q&A]
Is it possible to read the console output from DotNetCoreExecute?
for example to save to a file

**Comment by pascalberger** [comment]
You can use [StartProcess alias](https://cakebuild.net/dsl/process/) to run `dotnet` where you can pass settings to redirect standard error:

```csharp
IEnumerable<string> dotnetOutput;
var exitCode =
  StartProcess(
    "dotnet myassembly.dll",
    new ProcessSettings {
      RedirectStandardOutput = true,
    },
    out dotnetOutput);
```

**Reply by augustoproiete** [reply]
Tracking possible improvement for a future version of Cake here: https://github.com/cake-build/cake/issues/2461

Also relates to https://github.com/cake-build/cake/issues/2460

**Reply by BartNetJS** [reply]
I already tried it with StartProcess.
It was working on my local dev machine and when I used windows-latest in an azure devops pipeline also.
But with bash on ubuntu I had an issue with access denied on the dll. I tried a lot from the ubuntu community, but no success.
That was why I returned back to the cake extension DotNetCoreExecute

**Reply by augustoproiete** [reply]
Both `DotNetCoreExecute` and `StartProcess` call `Process.Start` to run the `dotnet` CLI, thus if one doesn't work for you, the other wouldn't either.

If you execute your build script with `--settings_showprocesscommandline=true` you can see what is the command being executed behind the scenes.

```powershell
dotnet cake --settings_showprocesscommandline=true
```

It seems to be an issue with the environment and/or the .NET installation. If you have repro steps, you can open an issue with Microsoft on https://github.com/dotnet/sdk

**Reply by BartNetJS** [reply]
that is strange...I will try it again
thanks for the suggestion

**Reply by augustoproiete** [reply]
@BartNetJS Did you get to the bottom of this one?
